src/util/simplify_expr_int.cpp

```diff
@@ -439,6 +445,8 @@ bool simplify_exprt::simplify_mod(exprt &expr)
 
 bool simplify_exprt::simplify_plus(exprt &expr)
 {
+  PRECONDITION(expr.id() == ID_plus);
```

Comment:
There's a `plus_exprt`, let's use this here.

Reply:
You mean `to_plus_exprt` instead of the `PRECONDITION`? I don't like that too much, the `PRECONDITION` here makes it clear we're expecting a `plus_exprt` whereas calling `to_plus_exprt` instead might be read as us just forgetting to check if it actually is...

Reply:
I was meaning to suggest that the function take a `plus_exprt &expr` as parameter. Will discuss further with `extractbit` below.